### PR TITLE
Rectify total number of Thompson MP species and add UFS-Aerosols to S2S and S2SW

### LINKED ISSUE #1114: Tracers improperly diffused in PBL if Thompson MP and prognostic aerosols are enabled
## Description
Tracers are improperly diffused in the PBL when both Thompson MP and prognostic aerosols are enabled in the atmospheric model. This is caused by a bug in CCPP-physics. See issue [#880](https://github.com/NCAR/ccpp-physics/issues/880) for details.

This bug affects prototype 8b.

### LINKED ISSUE #1117: Add UFS-Aerosols to S2S and S2SW build apps
## Description
Add UFS-Aerosols to the list of UFS components built using either `S2S` or `S2SW`.

### PR DESCRIPTION
**Pull request by rmontuoro**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Description

This PR includes the following changes:

1. It properly sets the number of species for the Thompson microphysics scheme in method [set_aerosol_tracer_index()](https://github.com/NCAR/ccpp-physics/blob/979324a89894133473c4663fc490d04cdce57ac0/physics/GFS_PBL_generic.F90#L14), used when both `trans_trac` and `cplchm` are `.true.`
2. It adds UFS-Aerosols to the UFS weather model build apps `S2S` and `S2SW`

Note that the baseline for the following regression test needs to be updated due to the bugfix introduced in 1.:
```
cpld_control_p8
cpld_2threads_p8
cpld_decomp_p8
cpld_mpi_p8
cpld_bmark_p8
cpld_control_c96_p8
cpld_control_c192_p8
cpld_control_c384_p8
cpld_debug_p8
```

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #1114
- fixes #1117

## Testing

These changes were tested on Hera/Intel.

Regression tests run on:

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [x] CI

## Dependencies

- waiting on https://github.com/NCAR/ccpp-physics/pull/883
- waiting on https://github.com/NOAA-EMC/fv3atm/pull/509

### CODE REVIEW

File: tests/rt.conf [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # S2S tests                                                                                                                                                                    #
 ###################################################################################################################################################################################
 
-COMPILE | -DAPP=S2SW -DUFS_GOCART=ON -DCCPP_SUITES=FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v16_coupled_p7_rrtmgp,FV3_GFS_v17_coupled_p8      | - wcoss_cray                            | fv3 |
+COMPILE | -DAPP=S2SW -DCCPP_SUITES=FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v16_coupled_p7_rrtmgp,FV3_GFS_v17_coupled_p8      | - wcoss_cray                            | fv3 |
```
Comment (junwang-noaa) [inline tests/rt.conf]: Can we use S2SWA for fv3+mom6+cice6+ww3+gocart application as we may want to keep s2sw for certain application such as GEFS ensemble members

Comment (junwang-noaa) [inline tests/rt.conf]: Can we use S2SWA for fv3+mom6+cice6+ww3+gocart application as we may want to keep s2sw for certain application such as GEFS ensemble members

**Comment by junwang-noaa** [comment]
@rmontuoro Your PR is the next in the commit queue. Would you mind adding one line documentation change for issue #1121 to your ufs-weather-model branch? There is no impact on the changes you made. 
```
        modified:   doc/UsersGuide/source/InputsOutputs.rst

[Jun.Wang@hfe04 source]$ git diff doc/UsersGuide/source/InputsOutputs.rst
@@ -1205,7 +1205,7 @@ shows the following parameters that can be set in *model_configure* at run-time.
    * - output_fh
      - history file output forecast hours or history file output frequency if the second elelment is -1
      - real
-     - -1 (negative: turn off the option, otherwise overwritten nfhout/nfhout_fh; 6 -1: output every 6 hoursr; 6 9: output history files at fh=6 and 9)
+     - -1 (negative: turn off the option, otherwise overwritten nfhout/nfhout_fh; 6 -1: output every 6 hoursr; 6 9: output history files at fh=6 and 9. Note: output_fh can only take 277 elements)

```
Thank you very much!

File: tests/rt.conf (hunk 1 above)
Comment (rmontuoro) [reply tests/rt.conf]: Done.

Comment (rmontuoro) [reply tests/rt.conf]: Done.

**Comment by rmontuoro** [comment]
> @rmontuoro Your PR is the next in the commit queue. Would you mind adding one line documentation change for issue #1121 to your ufs-weather-model branch? There is no impact on the changes you made.
> 
> ```
>         modified:   doc/UsersGuide/source/InputsOutputs.rst
> 
> [Jun.Wang@hfe04 source]$ git diff doc/UsersGuide/source/InputsOutputs.rst
> @@ -1205,7 +1205,7 @@ shows the following parameters that can be set in *model_configure* at run-time.
>     * - output_fh
>       - history file output forecast hours or history file output frequency if the second elelment is -1
>       - real
> -     - -1 (negative: turn off the option, otherwise overwritten nfhout/nfhout_fh; 6 -1: output every 6 hoursr; 6 9: output history files at fh=6 and 9)
> +     - -1 (negative: turn off the option, otherwise overwritten nfhout/nfhout_fh; 6 -1: output every 6 hoursr; 6 9: output history files at fh=6 and 9. Note: output_fh can only take 277 elements)
> ```
> 
> Thank you very much!

Done in 3347dddb312d9677426a4eae89733183bfcdc0f9.

**Comment by junwang-noaa** [comment]
@rmontuoro Thanks. The RT failed on wcoss because of the invalid application names. I will rerun it.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: BL
[BL] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/881736445/20220321123017/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/881736445/20220321131232/ufs-weather-model
[RT] Error: Test control_wam_debug 036 failed in run_test failed 
Please make changes and add the following label back: hera-gnu-BL

**Comment by junwang-noaa** [comment]
@rmontuoro The wcoss log files are on hera at: /scratch1/NCEPDEV/stmp2/Jun.Wang/wcosslog
Please commit to your branch. There are nodes not available on dell, the RT took longer than normal. Thanks

**Review by junwang-noaa — APPROVED** [review]

File: .gitmodules [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [submodule "FV3"]
   path = FV3
-  url = https://github.com/NOAA-EMC/fv3atm
-  branch = develop
+  url = https://github.com/rmontuoro/fv3atm
+  branch = bugfix/thompson-tracer-index
```
Comment (junwang-noaa) [inline .gitmodules]: @rmontuoro Please make an fv3atm PR, we need to merge your bugfix/thompson-tracer-index branch to emc develop branch.

**Comment by rmontuoro** [comment]
> @rmontuoro The wcoss log files are on hera at: /scratch1/NCEPDEV/stmp2/Jun.Wang/wcosslog
> Please commit to your branch. There are nodes not available on dell, the RT took longer than normal. Thanks

Done in f675eb0ac5f1df47b520bdf5a28269cbc64d2503.

File: .gitmodules (hunk 2 above)
Comment (rmontuoro) [reply .gitmodules]: See [PR#509](https://github.com/NOAA-EMC/fv3atm/pull/509).

**Comment by junwang-noaa** [comment]
ALL RT passed. The code is ready for commit. @grantfirl would you please commit CCPP PR#883? Thanks

**Comment by climbfuji** [comment]
Merged. New hash is [4a9b26c](https://github.com/NCAR/ccpp-physics/commit/4a9b26c6f53267e89ee2836e5295a32dfc03016e)

**Comment by junwang-noaa** [comment]
@rmontuoro Would you please update your fv3atm branch to point to ccpp physics main branch with revision [4a9b26c](https://github.com/NCAR/ccpp-physics/commit/4a9b26c6f53267e89ee2836e5295a32dfc03016e)? We can't merge your fv3 PR until you update it. Thanks

**Comment by rmontuoro** [comment]
> @rmontuoro Would you please update your fv3atm branch to point to ccpp physics main branch with revision [4a9b26c](https://github.com/NCAR/ccpp-physics/commit/4a9b26c6f53267e89ee2836e5295a32dfc03016e)? We can't merge your fv3 PR until you update it. Thanks

Done.

**Comment by junwang-noaa** [comment]
@rmontuoro Your fv3atm PR is merged, please update your ufs-weather-model branch to point to the latest fv3atm develop branch, revision [2b988fe](https://github.com/NOAA-EMC/fv3atm/commit/2b988fe191cc0dd9bf17f447a48654925f2c3509). Thanks

**Pull request merged**